src/vdc_collection.rs

```diff
@@ -0,0 +1,205 @@
+
+/// Supported credential formats.
+#[derive(PartialEq)]
+pub enum CredentialFormat {
+    MDoc,
+    W3cJwtVc,
+    W3cJwtVcJsonld,
+    Other(String),  // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String),  // For ease of expansion.
+}
+
+/// An individual credential.  These are mainly for internal use; access to them should go through `VdcCollection`,
+/// below.
+struct Credential {
+    id:      String,           // Probably a UUID, perhaps we should make it one?
+    format:  CredentialFormat,
+    ctype:   CredentialType,
+    tags:    Vec<String>,      // Arbitrary string tags.
+    payload: Vec<u8>,          // The actual credential.
+    dirty:   bool,             // Has this been modified since it was read from storage?
+}
+
+impl Credential {
+    /// Create a new credential.
+    fn new(
+        id:      &str,
+        format:  CredentialFormat,
+        ctype:   CredentialType,
+        payload: Vec<u8>,
+        dirty:   bool,
+    ) -> Credential {
+        Credential {
+            id:     id.to_string(),
+            format,
+            ctype,
+            tags:   Vec::new(),
+            payload,
+            dirty,
+        }
+    }
+
+    /// Check whether a credential has a specific tag.
+    fn contains_tag(&self, tag: &str) -> bool {
+        for t in self.tags.iter() {
+            if t == tag {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    /// Add a tag to a credential.
+    fn add_tag(&mut self, tag: &str) {
+        if !self.contains_tag(tag) {
+            self.tags.push(tag.to_string());
+            self.dirty = true;
+        }
+    }
```

Comment:
This isn't needed for this implementation, so could you remove this, so we can treat Credentials as immutable and then we don't have to worry about consumers of this API not saving "dirty" Credentials.

Reply:
Removed.